lib/cc/cli/source_fingerprint.rb

```diff
@@ -0,0 +1,69 @@
+require "digest/md5"
+
+module CC
+  module CLI
+    class SourceFingerprint
+      def initialize(output)
+        @output = output
+      end
+
+      def compute
+        md5 = Digest::MD5.new
+        md5 << key
```

Comment:
Nice! I like the work here and that you broke out the fingerprint computations into two separate functions, so that the old mechanism's still available. 

One question: Does this mean that anytime content changes the fingerprint will change? Could that create negative behavior changes in reporting new issues instead of `worse` issues - in fact perhaps not allowing issues to get worse in some cases? 

I suppose engines can implement their own fingerprints to account for some of those cases - like method length.